File: rb/features.rb
# Nextbike SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module NextbikeFeatures
  def self.make_feature(name)
    case name
    when "base"
      NextbikeBaseFeature.new
    when "ratelimit"
      NextbikeRatelimitFeature.new
    when "retry"
      NextbikeRetryFeature.new
    when "test"
      NextbikeTestFeature.new
    when "timeout"
      NextbikeTimeoutFeature.new
    else
      NextbikeBaseFeature.new
    end
  end
end

File: py/features.py
# Nextbike SDK feature factory

from feature.base_feature import NextbikeBaseFeature
from feature.test_feature import NextbikeTestFeature


def _make_feature(name):
    features = {
        "base": lambda: NextbikeBaseFeature(),
        "test": lambda: NextbikeTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
